File: src/dataset/javascript/bob_exercism-ThomasZumsteg/openAI/ChatGPT4_bob_v3.js
class Bob {
  hey(input) {
    const trimmedInput = input.trim();
    if (trimmedInput === '') return "Fine. Be that way!";
    if (trimmedInput === trimmedInput.toUpperCase() && /[A-Z]/.test(trimmedInput)) return "Whoa, chill out!";
    if (trimmedInput.endsWith('?')) return "Sure.";
    return "Whatever.";
  }
}

module.exports = Bob;

File: src/dataset/Java/bob_exercism-ThomasZumsteg/openAI/ChatGPT4_Bob_v3.java
public class Bob {
    public String hey(String greeting) {
        if (greeting == null || (greeting = greeting.trim()).isEmpty()) {
            return "Fine. Be that way!";
        }
        if (shouting(greeting)) {
            return "Whoa, chill out!";
        }
        if (greeting.endsWith("?")) {
            return "Sure.";
        }
        return "Whatever.";
    }

    private static boolean shouting(String greeting) {
        boolean hasUppercase = false;
        for (int i = 0, len = greeting.length(); i < len; i++) {
            char c = greeting.charAt(i);
            if (Character.isLowerCase(c)) {
                return false;
            }
            if (Character.isUpperCase(c)) {
                hasUppercase = true;
            }
        }
        return hasUppercase;
    }
}